pretyping/constr_matching.ml

```diff
@@ -308,7 +308,12 @@ let matches_core env sigma convert allow_partial_app allow_bound_rels
 
       |	PFix c1, Fix _ when eq_constr (mkFix c1) cT -> subst
       |	PCoFix c1, CoFix _ when eq_constr (mkCoFix c1) cT -> subst
-      | _ -> raise PatternMatchingFailure
+
+      | PEvar _, _ -> raise PatternMatchingFailure (* Is this expected? *)
```

Comment:
@ppedrot so what should this clause be? I understand too little of this code to write it myself.